flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/MySqlToElasticsearchE2eITCase.java

```diff
@@ -0,0 +1,378 @@
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.fail;
+
+/** End-to-end tests for mysql cdc to Elasticsearch pipeline job. */
+@RunWith(Parameterized.class)
+public class MySqlToElasticsearchE2eITCase extends PipelineTestEnvironment {
+    private static final Logger LOG = LoggerFactory.getLogger(MySqlToElasticsearchE2eITCase.class);
+
+    // ------------------------------------------------------------------------------------------
+    // MySQL Variables (we always use MySQL as the data source for easier verifying)
+    // ------------------------------------------------------------------------------------------
+    protected static final String MYSQL_TEST_USER = "mysqluser";
+    protected static final String MYSQL_TEST_PASSWORD = "mysqlpw";
+
+    @Parameterized.Parameter(1)
+    public String elasticsearchVersion;
+
+    public static final Duration DEFAULT_RESULT_VERIFY_TIMEOUT = Duration.ofSeconds(120);
+
+    @ClassRule
+    public static final MySqlContainer MYSQL =
+            (MySqlContainer)
+                    new MySqlContainer(
+                                    MySqlVersion.V8_0) // v8 support both ARM and AMD architectures
+                            .withConfigurationOverride("docker/mysql/my.cnf")
+                            .withSetupSQL("docker/mysql/setup.sql")
+                            .withDatabaseName("flink-test")
+                            .withUsername("flinkuser")
+                            .withPassword("flinkpw")
+                            .withNetwork(NETWORK)
+                            .withNetworkAliases("mysql");
+
+    private org.testcontainers.elasticsearch.ElasticsearchContainer elasticsearchContainer;
+    private static final Integer MAX_ELASTICSERACH_FETCH_SIZE = 20;
+    protected final UniqueDatabase mysqlInventoryDatabase =
+            new UniqueDatabase(MYSQL, "mysql_inventory", MYSQL_TEST_USER, MYSQL_TEST_PASSWORD);
+
+    @Parameterized.Parameters(name = "flinkVersion: {0}, elasticsearchVersion: {1}")
+    public static Collection<Object[]> getTestParameters() {
+        List<Object[]> parameters = new ArrayList<>();
+        for (String flinkVersion : getFlinkVersion()) {
+            parameters.add(new Object[] {flinkVersion, "6.8.20"});
+            parameters.add(new Object[] {flinkVersion, "7.10.2"});
+            parameters.add(new Object[] {flinkVersion, "8.12.1"});
+        }
+
+        return parameters;
+    }
+
+    @Before
+    public void before() throws Exception {
+        super.before();
+        mysqlInventoryDatabase.createAndInitialize();
+        elasticsearchContainer =
+                new ElasticsearchContainer(elasticsearchVersion)
+                        .withNetwork(NETWORK)
+                        .withNetworkAliases("elasticsearch")
+                        .withLogConsumer(new Slf4jLogConsumer(LOG));
+
+        Startables.deepStart(Stream.of(MYSQL, elasticsearchContainer)).join();
```

Comment:
Put these in `@BeforeClass` hooks? Putting it here seems a little too late (since we're accessing MySQL database previously)